ChemistryLib/PhreeqcIO.cpp

```diff
@@ -198,9 +259,20 @@ void PhreeqcIO::writeInputsToFile(double const dt)
 
 std::ostream& operator<<(std::ostream& os, PhreeqcIO const& phreeqc_io)
 {
-    os << "SELECTED_OUTPUT" << "\n";
+    auto const& knobs = phreeqc_io._knobs;
+    if (knobs)
+    {
+        os << *knobs << "\n";
+    }
+
     os << *phreeqc_io._output << "\n";
 
+    auto const& user_punch = phreeqc_io._user_punch;
+    if (user_punch)
```

Comment:
looks like `user_punch` is a pointer. If so, it would be better to store it by value. In the current implementation you have one more indirection.
Store small objects by value...

Reply:
Fixed. I will bear this criterion in mind.

Reply:
_user_punch is wrapped by `unique_ptr`...